### PR TITLE
FIX Explicitly use host Python version when building packages

### PR DESCRIPTION
**Pull request by ryanking13**
This change prevents the use of unintended Python versions when building packages.

Related: #1899

### CODE REVIEW

File: packages/Makefile [hunk 1]
```diff
@@ -1,6 +1,7 @@
 export PYODIDE_ROOT=$(abspath ..)
 PYODIDE_LIBRARIES=$(abspath ./.artifacts)
 CPYTHONROOT=$(PYODIDE_ROOT)/cpython
+HOSTPYTHON=$${HOSTPYTHON:-python3.9}
```
Comment (rth) [inline packages/Makefile:4]: It's already defined in Makefile.envs I believe

https://github.com/pyodide/pyodide/blob/b205f8e6bd176b02e701407099843aa8900fe123/Makefile.envs#L19

Why would we need to re-define it here, particularly before including the Makefile.envs ?

**Review by rth — COMMENTED** [review]
Minor comment, otherwise LGTM. Thanks @ryanking13 !

**Comment by ryanking13** [comment]
> Why would we need to re-define it here, particularly before including the Makefile.envs ?

Oh, thanks for pointing that out. It was for a test but I forgot to remove it before the commit. Removed it.

**Comment by ryanking13** [comment]
Hmm... It looks like the numpy build is failing in Github actions.

**Comment by rth** [comment]
Due to setuptools I think: https://github.com/pyodide/pyodide/issues/1930

**Comment by ryanking13** [comment]
> Due to setuptools I think: #1930

Indeed. https://github.com/numpy/numpy/pull/19173#issuecomment-960595766

**Comment by rth** [comment]
Could you also add me and Hood as collaborators to your fork CI https://app.circleci.com/pipelines/github/ryanking13/pyodide/ so we can restart jobs? In particular, it's strange that there are so many firefox package test failures and I wanted to restart it.

**Comment by ryanking13** [comment]
> Could you also add me and Hood as collaborators to your fork CI https://app.circleci.com/pipelines/github/ryanking13/pyodide/ so we can restart jobs? In particular, it's strange that there are so many firefox package test failures and I wanted to restart it.

I am having trouble finding a way to add a collaborator to my fork CI. Do you know how?
Anyway, I've restarted CI.

https://app.circleci.com/pipelines/github/ryanking13/pyodide?branch=packaging%2Fpython-version

**Comment by rth** [comment]
> I guess giving write permission to my fork repository would give access to fork CI as well?

Yes, I think that was the solution. Anyway it was just a though it's not critical, it should get better with larger CircleCI runners.

**Pull request merged**